### PR TITLE
refactor!: Rework to remove use of TOML package

### PR DESCRIPTION
**Pull request by lenny-intel**
BREAKING CHANGE: OverrideTomlValues changed to OverrideConfigMapValues

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A**
  <link to docs PR>

## Testing Instructions
See [edgex-go PR](https://github.com/edgexfoundry/edgex-go/pull/4468) testing instructions

### CODE REVIEW

File: bootstrap/config/config.go [hunk 1]
```diff
@@ -487,7 +479,7 @@ func CreateProviderClient(
 
 	// The passed in configStem already contains the trailing '/' in most cases so must verify and add if missing.
 	if configStem[len(configStem)-1] != '/' {
```
Comment (jim-wang-intel) [inline bootstrap/config/config.go:481]: should this also change to use that constant `pathSep`?

Comment (lenny-intel) [reply bootstrap/config/config.go:481]: It is a character , not a string so didn't change it.

Comment (jim-wang-intel) [reply bootstrap/config/config.go:481]: ok

**Review by jim-wang-intel — APPROVED** [review]
LGTM

File: bootstrap/config/config.go [hunk 2]
```diff
@@ -402,24 +400,28 @@ func (cp *Processor) getAccessTokenCallback(serviceKey string, secretProvider in
...
 		cp.lc.Info("Skipping use of Configuration Provider for custom configuration: Provider not available")
-		tomlTree, err := cp.loadPrivateFromFile()
+		filePath := GetConfigFileLocation(cp.lc, cp.flags)
+		configMap, err := cp.loadConfigYamlFromFile(filePath)
 		if err != nil {
 			return err
 		}
-		if err := tomlTree.Unmarshal(config); err != nil {
-			return fmt.Errorf("could not load toml tree into custom config: %s", err.Error())
+
```
Comment (ejlee3) [inline bootstrap/config/config.go:416]: can we use the convertMapToInterface function here?

File: bootstrap/config/config.go [hunk 3]
```diff
@@ -572,70 +564,46 @@ func CreateProviderClient(
...
 	err = yaml.Unmarshal(contents, &data)
 	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshall common configuration file %s: %s", yamlFile, err.Error())
+		return nil, fmt.Errorf("failed to unmarshall configuration file %s: %s", yamlFile, err.Error())
 	}
 	return data, nil
 }
 
-func (cp *Processor) mergeTomlWithConfig(config interface{}, tomlTree *toml.Tree) error {
+func (cp *Processor) mergeMapWithConfig(config interface{}, configMap map[string]any) error {
```
Comment (ejlee3) [inline bootstrap/config/config.go:584]:
```suggestion
func (cp *Processor) mergeMapWithConfig(config any, configMap map[string]any) error {
```

File: bootstrap/environment/variables.go [hunk 4]
```diff
@@ -186,7 +246,7 @@ func (e *Variables) buildPaths(keyMap map[string]interface{}) []string {
 
 		subPaths := e.buildPaths(subMap)
 		for _, path := range subPaths {
-			paths = append(paths, fmt.Sprintf("%s.%s", key, path))
+			paths = append(paths, fmt.Sprintf("%s/%s", key, path))
```
Comment (ejlee3) [inline bootstrap/environment/variables.go:249]:
```suggestion
			paths = append(paths, fmt.Sprintf("%s%s%s", key, configPathSeparator, path))
```

**Review by ejlee3 — CHANGES_REQUESTED** [review]

File: bootstrap/config/config.go (hunk 2 above)
Comment (lenny-intel) [reply bootstrap/config/config.go:417]: For which part? 
i.e. this comment is for blank line 416.

Comment (lenny-intel) [reply bootstrap/config/config.go:417]: ahh, ok. Thinking...

Comment (lenny-intel) [reply bootstrap/config/config.go:417]: Yep, thanks!

Comment (lenny-intel) [reply bootstrap/config/config.go:417]: reworked

File: bootstrap/config/config_test.go [hunk 5]
```diff
@@ -296,89 +297,6 @@ func TestLoadCommonConfigFromFile(t *testing.T) {
...
-	mergeMaps(destMap, srcMap)
-
-	for key, value := range destMap {
-		if key == "StoreAndForwardInfo" || key == "Trigger" {
-			assert.NotEmpty(t, value)
-		}
-	}
-}
-
 func TestRemoveZeroValues(t *testing.T) {
```
Comment (ejlee3) [inline bootstrap/config/config_test.go:300]: move this test too!

**Review by ejlee3 — APPROVED** [review]
LGTM! Thanks for doing the refactoring!

**Pull request merged**